tests/test_server.py

```diff
@@ -67,8 +75,74 @@ def test_duplicate_clientid(flask_client, fake_fs, state_init):
     config.reload()
     params = MultiDict(
         [
-            ("clientid", "client0"),
-            ("clientid", "client1"),
+            ("clientid", "a0eebc99-9c0b-4ef8-bb6d-6bb9bd380a11"),
+            ("key", "jk3o4g5j34h0"),
+            ("clientid", "123e4567-e89b-12d3-a456-426614174001"),
+            ("key", "jk3h4g5j34h0"),
         ]
     )
-    assert flask_client.post("/alive", query_string=params).status_code == 400
+    response = flask_client.post("/alive", query_string=params)
+    assert response.status_code == 400
+    assert 'Parameters "clientid" and "key" should be provided exactly once.' in response.get_data(
+        as_text=True
+    )
+
+
+def test_invalid_uuid_format(flask_client, fake_fs, state_init):
+    response = flask_client.post(
+        "/alive", query_string={"clientid": "invalid-uuid", "key": "jk3h4g5j34h0"}
+    )
+    assert response.status_code == 400
+    assert "not a valid UUID" in response.get_data(as_text=True)
```

Comment:
Comparing error messages against string literal make slightly brittle tests.
If the exact phrasing isn't important, perhaps asserting on the status code is enough? And maybe that len(response) > 0?

Reply:
I had a previous recommendation to not use UUIDs at all, if we do that, this code will not be needed anyway.